### ISSUE TITLE
Update the tech-docs-gem to the newest version of GOV.UK Frontend

### ISSUE THREAD

**Issue by vanitabarrett** [alphagov/govuk-design-system#1641]
## What
Update the tech-docs-gem to use version 3.12.0 of GOV.UK Frontend.

## Why
The [tech-docs-gem](https://github.com/alphagov/tech-docs-gem) is using version 3.6.0 of GOV.UK Frontend, which is a couple of minor releases behind. We will be releasing the new link styles in version 3.12.0 and think this will be a good time to update the tech-docs-gem to use this version so everyone benefits from the new link styles.

## Who needs to know about this
Developers

## Done when

- [x] The tech docs gem is using GOV.UK Frontend 3.12.0
- [x] The tech docs gem is opted into the new link styles
- [x] We should probably check the changelog between 3.6.0 and 3.12.0 to see if there are any other changes we need to make in the tech docs gem
- [x] Roll this change out to Frontend Docs?

**Comment by lfdebrux** [comment]
I've opened a PR to update GOV.UK Frontend in the tech-docs-gem repo: alphagov/tech-docs-gem#232. This PR includes updating the link styles to use the new styles and mixins in v3.12.0 (thanks for the help with that @36degrees), and also changes the Sass imports to take advantage of the quicker CSS compiles from v3.7.0.

Once that is merged I will be able to update the gem itself by opening another PR to bump the version and do a release.

**Comment by lfdebrux** [comment]
PR to release new version of tech docs gem https://github.com/alphagov/tech-docs-gem/pull/234

**Comment by lfdebrux** [comment]
[Ruby gem govuk_tech_docs v2.3.0](https://rubygems.org/gems/govuk_tech_docs/versions/2.3.0) has been released 🎉 Going to update frontend docs now.

**Comment by lfdebrux** [comment]
Updated frontend docs, new link styles are live at frontend.design-system.service.gov.uk. Calling this done.
